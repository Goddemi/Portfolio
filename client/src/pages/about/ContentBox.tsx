import React from "react";
import styled from "styled-components";

type ContentType = {
  keyName?: string;
  value?: string | JSX.Element;
};

interface Props {
  variable: string;
  content: ContentType[];
}

const ContentBox: React.FC<Props> = ({ variable, content }) => {
  return (
    <Container>
      <Declare>const</Declare>
      <Variable> {variable} </Variable>
      <Bracket>= {variable === "Five_Keywords" ? "[" : "{"} </Bracket>
      {content.map(({ keyName, value }) => {
        return (
          <Content>
            {variable !== "Five_Keywords" ? (
              <>
                <Key>&nbsp;&nbsp; {keyName}&nbsp;:&nbsp;</Key>
                <Value> {value}</Value>
                <Key>,</Key>
              </>
            ) : (
              <>
                <Value> &nbsp;&nbsp; {value}</Value>
                <Key>,</Key>
              </>
            )}
          </Content>
        );
      })}
      <Bracket>{variable === "Five_Keywords" ? "]" : "}"}</Bracket>
    </Container>
  );
};

export default ContentBox;

const Container = styled.div`
  width: 550px;
  height: 300px;
  margin: 10px 30px;
  padding: 20px;
  border: 1px solid gray;
  border-radius: ${(props) => props.theme.radius};
  span {
    font-size: 18px;
  }
`;

const Declare = styled.span`
  color: ${(props) => props.theme.purple};
`;

const Variable = styled.span`
  color: white;
`;

const Bracket = styled.span`
  color: ${(props) => props.theme.purple};
`;

const Content = styled.div`
  margin: 10px 0;
`;

const Key = styled.span`
  color: gray;
`;

const Value = styled.span`
  line-height: 25px;
  color: ${(props) => props.theme.green};
`;
