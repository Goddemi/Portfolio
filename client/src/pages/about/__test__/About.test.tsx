import renderer from "react-test-renderer";
import ContentBox from "../ContentBox";

describe("About Page", () => {
  test("About Contents render correctly", () => {
    const [variable, content] = ["test", [{ keyName: "이름", value: "test" }]];
    const tree = renderer
      .create(<ContentBox variable={variable} content={content} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
