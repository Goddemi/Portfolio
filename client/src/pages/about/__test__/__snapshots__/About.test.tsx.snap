// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page About Contents render correctly 1`] = `
<div
  className="sc-bcXHqe wPqGi"
>
  <span
    className="sc-gswNZR gRfICY"
  >
    const
  </span>
  <span
    className="sc-dkrFOg"
  >
     
    test
     
  </span>
  <span
    className="sc-hLBbgP cRWFli"
  >
    = 
    {
     
  </span>
  <div
    className="sc-eDvSVe fpQGor"
  >
    <span
      className="sc-jSUZER fkfKUW"
    >
         
      이름
       : 
    </span>
    <span
      className="sc-gKPRtg leLxhg"
    >
       
      test
    </span>
    <span
      className="sc-jSUZER fkfKUW"
    >
      ,
    </span>
  </div>
  <span
    className="sc-hLBbgP cRWFli"
  >
    }
  </span>
</div>
`;
